### PR TITLE
Network peering

### LINKED ISSUE #29: Bug VPC Peering between GCP and Atlas
Good morning,

I try to create peering between MongoDB Atlas Project and my VPC in GCP. 

```
resource "mongodbatlas_network_peering" "main_peering" {
  project_id        = mongodbatlas_project.project.id
  container_id      = mongodbatlas_network_container.container.container_id
  provider_name     = "GCP"
  gcp_project_id    = var.gcp_project_id
  network_name      = var.gcp_main_network_name
  depends_on       = ["mongodbatlas_network_container.container"]
}

resource "google_compute_network_peering" "gcp_main_atlas_peering" {
  name = "atlas-gcp-main"
  network = var.gcp_main_network_link
 peer_network="projects/${mongodbatlas_network_peering.main_peering.project_id}/global/networks/${mongodbatlas_network_peering.main_peering.network_name}"
}
```

But It never finish. I try it several times but it always finished with the same error:  

```
module.atlas.mongodbatlas_network_peering.main_peering: Still creating... [45m3s elapsed]

Error: rpc error: code = Unavailable desc = transport is closing
```

In the Atlas website appear this message:
**Create a reciprocal connection from GCP to Atlas to complete this connection. Show me how**

Thank you,

### PR DESCRIPTION
**Pull request by PacoDw**
- Added `atlas_gcp_project_id` and `atlas_vpc_name` to peering state to be more easy the flow.
- The documentation was updated.
- The GCP test case was updated.

You can use the following configuration to create the reciprocal connection:

```hcl
locals {
  project_id        = <your-project-id>
  google_project_id = <your-google-project-id>
}

resource "mongodbatlas_network_container" "test" {
  project_id       = local.project_id
  atlas_cidr_block = "192.168.192.0/18"
  provider_name    = "GCP"
}

resource "mongodbatlas_private_ip_mode" "my_private_ip_mode" {
  project_id = local.project_id
  enabled    = true
}

resource "mongodbatlas_network_peering" "test" {
  project_id     = local.project_id
  container_id   = mongodbatlas_network_container.test.container_id
  provider_name  = "GCP"
  network_name   = "myNetWorkPeering"
  gcp_project_id = local.google_project_id

  depends_on = [mongodbatlas_private_ip_mode.my_private_ip_mode]
}

resource "google_compute_network" "vpc_network" {
  name = "vpcnetwork"
}

resource "google_compute_network_peering" "gcp_main_atlas_peering" {
  name         = "atlas-gcp-main"
  network      = google_compute_network.vpc_network.self_link
  peer_network = "projects/${mongodbatlas_network_peering.test.atlas_gcp_project_id}/global/networks/${mongodbatlas_network_peering.test.atlas_vpc_name}"
}
```
closes #29 , #41

### CODE REVIEW

**Review by marinsalinas — CHANGES_REQUESTED** [review]
Could you check the `make lint` issue?

**Comment by PacoDw** [comment]
@marinsalinas this appears:
```
==> Checking source code against linters...
/bin/sh: golangci-lint: command not found
make: *** [lint] Error 127
```

Maybe is regarding this issue: https://github.com/golangci/golangci-lint/issues/308

**Comment by marinsalinas** [comment]
> @marinsalinas this appears:
> 
> ```
> ==> Checking source code against linters...
> /bin/sh: golangci-lint: command not found
> make: *** [lint] Error 127
> ```

you should run `make tools` before.

The lint error is the following: 

```log
==> Checking source code against linters...
mongodbatlas/resource_mongodbatlas_network_container.go:22:2: `errorContainerDelete` is unused (varcheck)
	errorContainerDelete  = "error deleting MongoDB Network Peering Container (%s): %s"

```

**Review by marinsalinas — APPROVED** [review]
LGTM! Waiting for @themantissa  review to merge it!

**Review by themantissa — APPROVED** [review]
Approving (if my slack comment to you about how this flows is right).  It's  a very nice way to handle this!

**Pull request merged**
